Mathlib/Analysis/SpecialFunctions/Integrability/LogMeromorphic.lean

```diff
@@ -0,0 +1,161 @@
+      ((MeromorphicOn.divisor f [[a, b]]).finiteSupport isCompact_uIcc)
+    have h₄g := MeromorphicOn.extract_zeros_poles_log h₂g h₃g
+    rw [intervalIntegrable_congr_codiscreteWithin
+      (h₄g.filter_mono (Filter.codiscreteWithin.mono Set.uIoc_subset_uIcc))]
+    apply IntervalIntegrable.add
+    · apply IntervalIntegrable.finsum
+      intro i
+      apply IntervalIntegrable.const_mul
+      rw [(by ring : a = ((a - i) + i)), (by ring : b = ((b - i) + i))]
+      apply IntervalIntegrable.comp_sub_right (f := (log ‖·‖)) _ i
+      simp [norm_eq_abs, log_abs]
+    · apply ContinuousOn.intervalIntegrable
+      apply h₁g.continuousOn.norm.log
+      simp_all
+  · rw [← hf.exists_meromorphicOrderAt_ne_top_iff_forall (isConnected_Icc inf_le_sup)] at t₀
+    push_neg at t₀
+    have : (log ‖f ·‖) =ᶠ[Filter.codiscreteWithin (Ι a b)] 0 := by
+      apply Filter.EventuallyEq.filter_mono _ (Filter.codiscreteWithin.mono Set.uIoc_subset_uIcc)
+      filter_upwards [hf.meromorphicNFAt_mem_codiscreteWithin,
+        Filter.self_mem_codiscreteWithin [[a, b]]] with x h₁x h₂x
+      simp only [Pi.zero_apply, log_eq_zero, norm_eq_zero]
+      left
+      by_contra hCon
+      simp_all [← h₁x.meromorphicOrderAt_eq_zero_iff, t₀ ⟨x, h₂x⟩]
+    rw [intervalIntegrable_congr_codiscreteWithin this]
+    apply _root_.intervalIntegrable_const_iff.2
+    tauto
+
+/--
+If `f` is real-meromorphic on a compact interval, then `log ‖f ·‖` is interval integrable on this
+interval.
+-/
+theorem intervalIntegrable_posLog_norm_meromorphicOn (hf : MeromorphicOn f [[a, b]]) :
+    IntervalIntegrable (log⁺ ‖f ·‖) volume a b := by
+  simp_rw [← half_mul_log_add_log_abs, mul_add]
+  apply IntervalIntegrable.add
+  · apply (intervalIntegrable_log_norm_meromorphicOn hf).const_mul
+  · apply (intervalIntegrable_log_norm_meromorphicOn hf).abs.const_mul
+
+/--
+If `f` is real-meromorphic on a compact interval, then `log ∘ f` is interval integrable on this
+interval.
+-/
+theorem _root_.MeromorphicOn.intervalIntegrable_log {f : ℝ → ℝ} (hf : MeromorphicOn f [[a, b]]) :
+    IntervalIntegrable (log ∘ f) volume a b := by
+  rw [(by aesop : log ∘ f = (log ‖f ·‖))]
+  exact intervalIntegrable_log_norm_meromorphicOn hf
+
+/--
+Special case of `MeromorphicOn.intervalIntegrable_log`: The function `log ∘ sin` is interval
+integrable over every interval.
+-/
+theorem intervalIntegrable_log_sin : IntervalIntegrable (log ∘ sin) volume a b :=
+  analyticOnNhd_sin.meromorphicOn.intervalIntegrable_log
+
+end intervalIntegral
+
+/-!
+## Circle Integrability for Logarithms of Complex Meromorphic Functions
+-/
+
+namespace circleIntegral
+
+variable
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace ℂ E]
+  {c : ℂ} {R : ℝ} {f : ℂ → E}
+
+/--
+If `f` is complex meromorphic on a circle in the complex plane, then `log ‖f ·‖` is circle
+integrable over that circle.
+-/
+theorem circleIntegrable_log_norm_meromorphicOn (hf : MeromorphicOn f (sphere c |R|)) :
```

Comment:
I would guess that in 95% of applications, R will be nonnegative, and the assumption will be of the form `MeromorphicOn f (sphere c R)`. Could you also give the statement in this form, called `circleIntegrable_log_norm_meromorphicOn_of_nonneg`?

Reply:
Done.